### PR TITLE
feature: Graviton support for XGB and SKLearn frameworks

### PR DESCRIPTION
**Pull request by mabunday**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by knikure — COMMENTED** [review]
/bot run slow-tests, pr

**Review by dewan-c — APPROVED** [review]

File: src/sagemaker/image_uris.py [hunk 1]
```diff
@@ -152,6 +155,12 @@ def retrieve(
             inference_tool = _get_inference_tool(inference_tool, instance_type)
             if inference_tool == "neuron":
                 _framework = f"{framework}-{inference_tool}"
+        # Preserves backwards compatibility with XGB/SKLearn configs which no
+        # longer define top-level "scope" keys after introducing support for
+        # Graviton inference. Training and inference configs for XGB/SKLearn are
+        # identical, so default to training.
+        if image_scope is None and framework in (XGBOOST_FRAMEWORK, SKLEARN_FRAMEWORK):
+            image_scope = "training"
```
Comment (knikure) [inline src/sagemaker/image_uris.py:163]: Kindly create a function for this.

File: src/sagemaker/image_uris.py [hunk 2]
```diff
@@ -244,6 +253,20 @@ def retrieve(
         if key in container_versions:
             tag = "-".join([tag, container_versions[key]])
 
+    if framework in (XGBOOST_FRAMEWORK, SKLEARN_FRAMEWORK) and isinstance(instance_type, str):
```
Comment (knikure) [inline src/sagemaker/image_uris.py:256]: Kindly create a function for this too.

**Review by knikure — COMMENTED** [review]
/bot run slow-tests

**Review by knikure — APPROVED** [review]

**Review by navinsoni — APPROVED** [review]

**Pull request merged**